src/main/java/bio/terra/stairway/FlightResult.java

```diff
@@ -0,0 +1,50 @@
+package bio.terra.stairway;
+
+import org.apache.commons.lang3.builder.ToStringStyle;
+
+import java.util.Optional;
+
+/**
+ * Class that holds the result of the flight returned to the caller.
+ */
+public class FlightResult {
+    private boolean success;
+    private Optional<Throwable> throwable;
+    private SafeHashMap resultMap;
+
+
+    FlightResult(StepResult stepResult, SafeHashMap workingMap) {
+        this.success = (stepResult.getStepStatus() == StepStatus.STEP_RESULT_SUCCESS);
+        this.throwable = stepResult.getThrowable();
+        this.resultMap = workingMap;
+        this.resultMap.setImmutable();
+    }
+
+    // Make a failed fatal flight result with an exception
+    public static FlightResult flightResultFatal(Exception ex) {
+        return new FlightResult(
+                new StepResult(StepStatus.STEP_RESULT_FAILURE_FATAL, ex),
+                new SafeHashMap());
```

Comment:
I think we need the current state of the working map to be persisted as part of the fatal failure. especially if we need to do manual clean up.

Reply:
Good point. I think we should persist it as part of the log and not try to write it to the database, since that may cause the failure.